### DISCUSSION TITLE
Audio Host -- How painful is this gonna be :/

### DISCUSSION THREAD

**Discussion by NineBallo** [hathach/tinyusb#2141] [Q&A]
I want to implement some form of audio host for creating a BT receiver. For some specific reasons I need it to be bluetooth->microcontroller(pico w)->usb dac. 

Where should I start and how bad is this gonna be to get working... and has anyone started or tried this before?

**Comment by hathach** [comment]
If you haven't familliar with tinyusb host API, maybe trying to have some hand-on testing with existing host driver e.g cdc or msc first. To get the big picture of how the usbh manages drivers etc. Then start to work on with enumeration first, by parsing your DAC descriptor device. Audio is the huge specs, you don't have to support all the things, just start small with your device specifically to make sure it work first. It is a bumpy road for sure, host driver is slightly more complicated than device one.
